activerecord/lib/active_record/relation/query_attribute.rb

```diff
@@ -36,10 +36,24 @@ def unboundable?
         @_unboundable
       end
 
+      def identifier_value
+        if unboundable?
+          super
```

Comment:
Condition to deal with too large integer numbers that make the Integer serialization fail. See https://github.com/rails/rails/commit/31ffbf8d5056137717da3f11d28c4fbd7fbc8f07.